src/libguac/tcp.c

```diff
@@ -108,12 +111,46 @@ int guac_tcp_connect(const char* hostname, const char* port, const int timeout)
                 FD_SET(fd, &fdset);
                 
                 retval = select(fd + 1, NULL, &fdset, NULL, &tv);
+
+                if (retval > 0) {
+                    int so_error = 0;
+                    socklen_t so_error_len = sizeof(so_error);
+                    retval = getsockopt(fd, SOL_SOCKET, SO_ERROR, &so_error, &so_error_len);
+
+                    // Error occurs retrieving the socket status, so we set it
+                    // and move on.
```

Comment:
We should prefer `/* C-style block comments */` to `// C++-style comments` in C code (same elsewhere in these changes).